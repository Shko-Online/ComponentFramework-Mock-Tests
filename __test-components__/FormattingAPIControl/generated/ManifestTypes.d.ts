/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

// Define IInputs and IOutputs Type. They should match with ControlManifest.
export interface IInputs {
    currencyInput: ComponentFramework.PropertyTypes.NumberProperty;
    dateInput: ComponentFramework.PropertyTypes.DateTimeProperty;
    decimalInput: ComponentFramework.PropertyTypes.DecimalNumberProperty;
    integerInput: ComponentFramework.PropertyTypes.WholeNumberProperty;
}
export interface IOutputs {
    currencyInput?: number;
    dateInput?: Date;
    decimalInput?: number;
    integerInput?: number;
}
