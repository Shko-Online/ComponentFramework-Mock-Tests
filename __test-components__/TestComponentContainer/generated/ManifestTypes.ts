/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

// Define IInputs and IOutputs Type. They should match with ControlManifest.
export interface IInputs {
    controlValue: ComponentFramework.PropertyTypes.StringProperty;
}
export interface IOutputs {
    controlValue?: string;
}
