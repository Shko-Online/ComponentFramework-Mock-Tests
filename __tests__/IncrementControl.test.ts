/*
	Unless explicitly acquired and licensed from Licensor under another
	license, the contents of this file are subject to the Reciprocal Public
	License ("RPL") Version 1.5, or subsequent versions as allowed by the RPL,
	and You may not copy or use this file in either source code or executable
	form, except in compliance with the terms and conditions of the RPL.

	All software distributed under the RPL is provided strictly on an "AS
	IS" basis, WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESS OR IMPLIED, AND
	LICENSOR HEREBY DISCLAIMS ALL SUCH WARRANTIES, INCLUDING WITHOUT
	LIMITATION, ANY WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
	PURPOSE, QUIET ENJOYMENT, OR NON-INFRINGEMENT. See the RPL for specific
	language governing rights and limitations under the RPL. 
*/

import * as sinon from 'sinon';

import { ComponentFrameworkMockGenerator } from '@shko-online/componentframework-mock/ComponentFramework-Mock-Generator';
import { IncrementControl } from '@powerapps-samples/increment-control/IncrementControl';
import { IInputs, IOutputs } from '@powerapps-samples/increment-control/IncrementControl/generated/ManifestTypes';
import { NumberPropertyMock } from '@shko-online/componentframework-mock/ComponentFramework-Mock/PropertyTypes/NumberProperty.mock';

describe("IncrementControl", () => {
	let mockGenerator: ComponentFrameworkMockGenerator<IInputs,IOutputs>;
	beforeEach(()=>{
        const container = document.createElement('div');

         mockGenerator = new ComponentFrameworkMockGenerator(
            IncrementControl,
            {
                value: NumberPropertyMock
            },
            container);
        document.body.appendChild(container);
    }
    )

    afterEach(()=>{
        document.body.innerHTML = null;
    })
	it("Init should work", () => {            
        mockGenerator.ExecuteInit();
        sinon.assert.calledOnce(mockGenerator.control.init);
        expect(document.body).toMatchSnapshot();
    })
	it("Update View should Work", () => {    
        mockGenerator.ExecuteInit();
        mockGenerator.ExecuteUpdateView();
        sinon.assert.calledOnce(mockGenerator.control.init);
        sinon.assert.calledOnce(mockGenerator.control.updateView);
        expect(document.body).toMatchSnapshot();
    })
	it("Update View with value should work", () => {
		mockGenerator.context.parameters.value.raw =100;
		mockGenerator.ExecuteInit();
		mockGenerator.ExecuteUpdateView();
		sinon.assert.calledOnce(mockGenerator.control.init);
        sinon.assert.calledOnce(mockGenerator.control.updateView);
        expect(document.body).toMatchSnapshot();
	})
	it("Click should Work", () => {
		mockGenerator.context.parameters.value.raw =100;
		mockGenerator.ExecuteInit();
        mockGenerator.ExecuteUpdateView();
        expect(document.body).toMatchSnapshot();

		const button =mockGenerator.container.querySelector("button");
		var evt =document.createEvent("Event");
		evt.initEvent("click",true, false);
		button.dispatchEvent(evt);

		mockGenerator.ExecuteUpdateView();
		console.log(mockGenerator.control.getOutputs().value);
		expect(mockGenerator.control.getOutputs().value).toEqual(101);
		expect(document.body).toMatchSnapshot();
	
		
	})

});