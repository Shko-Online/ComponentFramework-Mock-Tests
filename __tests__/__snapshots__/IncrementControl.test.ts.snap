// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncrementControl Click should Work 1`] = `
<body>
  <div>
    <div>
      <input
        type="label"
      />
      <button
        class="SimpleIncrement_Button_Style"
      >
        Increment
      </button>
    </div>
  </div>
</body>
`;

exports[`IncrementControl Click should Work 2`] = `
<body>
  <div>
    <div>
      <input
        type="label"
      />
      <button
        class="SimpleIncrement_Button_Style"
      >
        Increment
      </button>
    </div>
  </div>
</body>
`;

exports[`IncrementControl Init should work 1`] = `
<body>
  <div>
    <div>
      <input
        type="label"
      />
      <button
        class="SimpleIncrement_Button_Style"
      >
        Increment
      </button>
    </div>
  </div>
</body>
`;

exports[`IncrementControl Update View should Work 1`] = `
<body>
  <div>
    <div>
      <input
        type="label"
      />
      <button
        class="SimpleIncrement_Button_Style"
      >
        Increment
      </button>
    </div>
  </div>
</body>
`;

exports[`IncrementControl Update View with value should work 1`] = `
<body>
  <div>
    <div>
      <input
        type="label"
      />
      <button
        class="SimpleIncrement_Button_Style"
      >
        Increment
      </button>
    </div>
  </div>
</body>
`;
